src/Core/Components/DateTime/FluentDatePicker.razor.cs

```diff
@@ -66,10 +66,15 @@ protected async Task OnSelectedDateAsync(DateTime? value)
     {
         Opened = false;
 
-        var updatedValue = Value?.TimeOfDay != TimeSpan.Zero
+        DateTime? updatedValue = value;
+
+        if (Value is not null && value is not null)
+        {
+            updatedValue = Value?.TimeOfDay != TimeSpan.Zero
             ? (value ?? DateTime.MinValue).Date + Value?.TimeOfDay
```

Comment:
Why do you check `value` with `??` if the previous test already that?

Reply:
my bad. I removed it :)